### PR TITLE
add: non-param effect sizes

### PR DESCRIPTION
**Pull request by mattansb**
@IndrajeetPatil this is on dev. Please give this your usual testing, see where it breaks (other than what we already know...), then I'll add tests and merge to main.

``` r
library(effectsize)
```

`wilcox.test`

``` r
A <- c(48, 48, 77, 86, 85, 85)
B <- c(14, 34, 34, 77)
rank_biserial(A, B)
#> Loading required namespace: boot
#> r (rank biserial) |       95% CI
#> --------------------------------
#> 0.83              | [0.38, 1.00]
effectsize(wilcox.test(A, B))
#> Warning in wilcox.test.default(A, B): cannot compute exact p-value with ties
#> r (rank biserial) |       95% CI
#> --------------------------------
#> 0.83              | [0.30, 1.00]

x <- c(1.83,  0.50,  1.62,  2.48, 1.68, 1.88, 1.55, 3.06, 1.30)
y <- c(0.878, 0.647, 0.598, 2.05, 1.06, 1.29, 1.06, 3.14, 1.29)
rank_biserial(x, y, paired = TRUE)
#> r (rank biserial) |         95% CI
#> ----------------------------------
#> -0.78             | [-1.00, -0.20]
effectsize(wilcox.test(x, y, paired = TRUE))
#> r (rank biserial) |         95% CI
#> ----------------------------------
#> -0.78             | [-1.00, -0.33]
```

`kruskal.test`

``` r
x1 <- c(2.9, 3.0, 2.5, 2.6, 3.2) # normal subjects
x2 <- c(3.8, 2.7, 4.0, 2.4)      # with obstructive airway disease
x3 <- c(2.8, 3.4, 3.7, 2.2, 2.0) # with asbestosis
x <- c(x1, x2, x3)
g <- factor(rep(1:3, c(5, 4, 5)))
rank_epsilon_squared(x, g)
#> Epsilon2 (rank) |       95% CI
#> ------------------------------
#> 0.06            | [0.01, 0.72]
effectsize(kruskal.test(x, g))
#> Epsilon2 (rank) |       95% CI
#> ------------------------------
#> 0.06            | [0.01, 0.70]
```

`friedman.test`

``` r
wb <- aggregate(warpbreaks$breaks,
                by = list(w = warpbreaks$wool,
                          t = warpbreaks$tension),
                FUN = mean)
kendalls_w(x ~ w | t, data = wb)
#> Kendall's W |       95% CI
#> --------------------------
#> 0.08        | [0.08, 0.75]
effectsize(friedman.test(wb$x, wb$w, wb$t))
#> Kendall's W |       95% CI
#> --------------------------
#> 0.08        | [0.08, 0.75]
```

<sup>Created on 2021-01-05 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### CODE REVIEW

**Comment by IndrajeetPatil** [comment]
Cool, will have a look at this by the end of the day!

**Comment by mattansb** [comment]
Please take longer 😅

**Review by IndrajeetPatil — COMMENTED** [review]
Although the numeric values match up pretty closely, the sign of the effect size is reversed. 
Is this expected?

``` r
set.seed(123)
as.data.frame(effectsize::rank_biserial(ToothGrowth$len, mu = 20, ci = 0.95))
#> Warning in ranktransform.numeric(d, sign = TRUE): Zeros detected. These cannot
#> be sign-rank transformed.
#> Loading required namespace: boot
#>   r_rank_biserial   CI      CI_low   CI_high
#> 1       0.1485876 0.95 -0.08878716 0.4240099

set.seed(123)
rcompanion::wilcoxonOneSampleRC(ToothGrowth$len, mu = 20, ci = TRUE)
#>       rc lower.ci upper.ci
#> 1 -0.149   -0.438    0.157
```

<sup>Created on 2021-01-05 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>

**Comment by IndrajeetPatil** [comment]
The reversing of sign issue is also observed for paired version of rank-biserial:

```r
# data
df <-
  structure(list(Bird = c(
    "A", "B", "C", "D", "E", "F", "G", "H",
    "I", "J", "K", "L", "M", "N", "O", "P", "A", "B", "C", "D", "E",
    "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P"
  ), type = c(
    "Typical",
    "Typical", "Typical", "Typical", "Typical", "Typical", "Typical",
    "Typical", "Typical", "Typical", "Typical", "Typical", "Typical",
    "Typical", "Typical", "Typical", "Odd", "Odd", "Odd", "Odd",
    "Odd", "Odd", "Odd", "Odd", "Odd", "Odd", "Odd", "Odd", "Odd",
    "Odd", "Odd", "Odd"
  ), length = c(
    -0.255, -0.213, -0.19, -0.185,
    -0.045, -0.025, -0.015, 0.003, 0.015, 0.02, 0.023, 0.04, 0.04,
    0.05, 0.055, 0.058, -0.324, -0.185, -0.299, -0.144, -0.027, -0.039,
    -0.264, -0.077, -0.017, -0.169, -0.096, -0.33, -0.346, -0.191,
    -0.128, -0.182
  )), row.names = c(NA, -32L), class = c(
    "tbl_df",
    "tbl", "data.frame"
  ))

set.seed(123)
rcompanion::wilcoxonPairedRC(x = df$length, g = df$type, ci = TRUE)
#>       rc lower.ci upper.ci
#> 1 -0.853       -1   -0.559

set.seed(123)
as.data.frame(effectsize::rank_biserial(x = length ~ type, data = df, paired = TRUE, ci = 0.95))
#>   r_rank_biserial   CI    CI_low CI_high
#> 1       0.8529412 0.95 0.6316176       1
```

**Comment by IndrajeetPatil** [comment]
Also, there seems to be a bug in `kendall_w`:

```r
library(effectsize)
library(statsExpressions)

friedman.test(value ~ condition | id, data = iris_long)
#> 
#>  Friedman rank sum test
#> 
#> data:  value and condition and id
#> Friedman chi-squared = 410, df = 3, p-value < 2.2e-16

any(is.na(iris_long))
#> [1] FALSE

set.seed(123)
as.data.frame(kendalls_w(value ~ condition | id, data = iris_long))
#> Error in friedman.test.default(x, groups, blocks): NA's are not allowed in 'groups' or 'blocks'
```

**Comment by IndrajeetPatil** [comment]
I think that's it from me for now! 😅

File: R/rank_effectsizes.R [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+#'
+#' @references
+#' - Cureton, E. E. (1956). Rank-biserial correlation. Psychometrika, 21(3), 287-290.
+#' - Kerby, D. S. (2014). The simple difference formula: An approach to teaching nonparametric correlation. Comprehensive Psychology, 3, 11-IT.
+#' - Tomczak, M., & Tomczak, E. (2014). The need to report effect size estimates revisited. An overview of some recommended measures of effect size.
+#'
+#' @export
+rank_biserial <- function(x, y = NULL, data = NULL, mu = 0,
+                          ci = 0.95, iterations = 200,
+                          paired = FALSE) {
```
[the remark this answers is not in the document]
Comment (mattansb) [reply R/rank_effectsizes.R:83]: Done.

File: R/rank_effectsizes.R [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+#'
+#' # Confidence Intervals
+#' Confidence Intervals are estimated using the bootstrap method.
+#'
+#' @return A data frame with the effect size (`r_rank_biserial`, `Kendalls_W` or
+#'   `rank_epsilon_squared`) and its CI (`CI_low` and `CI_high`).
+#'
+#' @family effect size indices
+#'
+#' @examples
```
[the remark this answers is not in the document]
Comment (mattansb) [reply R/rank_effectsizes.R:53]: Done.

File: R/rank_effectsizes.R [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+#'                 FUN = mean)
+#' kendalls_w(x ~ w | t, data = wb)
+#'
+#' @references
+#' - Cureton, E. E. (1956). Rank-biserial correlation. Psychometrika, 21(3), 287-290.
+#' - Kerby, D. S. (2014). The simple difference formula: An approach to teaching nonparametric correlation. Comprehensive Psychology, 3, 11-IT.
+#' - Tomczak, M., & Tomczak, E. (2014). The need to report effect size estimates revisited. An overview of some recommended measures of effect size.
+#'
+#' @export
+rank_biserial <- function(x, y = NULL, data = NULL, mu = 0,
```
[the remark this answers is not in the document]
Comment (mattansb) [reply R/rank_effectsizes.R:81]: Done

**Comment by mattansb** [comment]
Hmm kendall's W seems to be wrong. Will fix later.

**Comment by mattansb** [comment]
Will add tests later today / tomorrow - then can merge.

**Comment by IndrajeetPatil** [comment]
Thanks! Everything looks good now :)

**Pull request merged**